### DISCUSSION TITLE
Adding local plugin/ Changing firefox to chrome in client script

### DISCUSSION THREAD

**Discussion by Nikhilesh212** [dane-tool/dane#18] [Q&A] [answered]
Hey, this is Nikhilesh from Viasat India. We are working on developing a video QoE for our network. We are using DANE to emulate Viasat like network, for that we need to integrate a chrome plugin with DANE, is there any way you could help us?

**Comment by parkeraddison** [comment, marked as answer]
Hi Nikhilesh,

## To use Google Chrome + ChromeDriver

You'll need to change the Docker image to install these rather than Firefox+Geckodriver. We'll need to change these lines:
https://github.com/dane-tool/dane/blob/95b17038f8cfd18c09f097628657715862ece5fd/docker/client/Dockerfile#L22-L36

Here's an easy way of doing so:
```Dockerfile
# Install Google Chrome
RUN wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb && \
    apt-get update && \
    apt-get install ./google-chrome-stable_current_amd64.deb

# Install ChromeDriver
RUN wget https://chromedriver.storage.googleapis.com/$(curl https://chromedriver.storage.googleapis.com/LATEST_RELEASE)/chromedriver_linux64.zip && \
    unzip chromedriver_linux64.zip -d /usr/local/bin/
```

**(don't forget to run `make build` and set the `.system.use_local_images` field to `true` in the config file so that the new Docker image will be used!)**

## To use a Chrome extension

You'll need to download the extension as a `.crx` file and drop it in the same directory as your selenium script. Then in the script you can use
```python
driver.add_extension(<path to extension .crx file>)
```
to load in the extension.

We did this over at [dane-tool/starter-scripts](https://github.com/dane-tool/starter-scripts/) for a Firefox addon -- the addon file is in extensions/ and the [streaming/endless-youtube.py](https://github.com/dane-tool/starter-scripts/blob/main/streaming/endless_youtube.py#L28-L31) script gets the absolute path to the addon then installs it. If you place the extension in the same directory then you can avoid the need for `.parent` stuff.

Hope this helps.
